tests/rptest/services/redpanda.py

```diff
@@ -2259,6 +2259,71 @@ def assert_cluster_is_reusable(self) -> None:
             f'Expected {expected_nodes} per tier definition but there '
             f'were only {broker_count} brokers: {brokers}')
 
+    def detect_spikes_by_percentile(self,
+                                    results,
+                                    significant_factor=1.5,
+                                    max_spike_width=1,
+                                    mode="individual"):
+        """
+        Detect spikes in latency data by percentile
+        
+        Args:
+            results: A dictionary containing latency metrics
+            significant_factor: Factor used to determine the spike threshold
+            max_spike_width: Maximum width (in consecutive data points) for a spike to be considered short-term
+            mode: "individual" to check each latency type separately, "all" to check combined
+            
+        Returns:
+            True if spikes detected according to the specified mode, False otherwise
+        """
+        def detect_spikes_in_series(latency_series):
+            sorted_series = sorted(latency_series)
+            n = len(sorted_series)
+
+            # median calculation
+            if n % 2 == 0:
+                median_value = (sorted_series[n // 2 - 1] +
+                                sorted_series[n // 2]) / 2
+            else:
+                median_value = sorted_series[n // 2]
+
+            upper_bound = median_value * significant_factor
+
+            # Detect spikes within the series
+            consecutive_exceeds = 0
+            for value in latency_series:
+                if value > upper_bound:
+                    consecutive_exceeds += 1
+                    if consecutive_exceeds > max_spike_width:
+                        return False  # Long-term high latency detected, not considered a spike
+                else:
+                    if 0 < consecutive_exceeds <= max_spike_width:
+                        # Detected a short-term spike
+                        if mode == "individual":
+                            return True
+                        consecutive_exceeds = 0
+            return consecutive_exceeds > 0 if mode == "all" else False
+
+        # Check for spikes based on the mode
+        if mode == "all":
+            combined_series = sum((results.get(key, []) for key in [
+                "publishLatency95pct", "publishLatency99pct",
+                "publishLatency999pct"
+            ]), [])
+            return detect_spikes_in_series(combined_series)
+        elif mode == "individual":
+            for key in [
+                    "publishLatency95pct", "publishLatency99pct",
```

Comment:
minor comment: Would have this as a constant property. Something like
```
spike_percentiles = ["publishLatency95pct", "publishLatency99pct",
                     "publishLatency999pct"]
```

Reply:
Removed this part, we get benchmark.metrics directly from results.json and latency series are specified in omb_validation_test

Thank you for suggestion